external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaRelationSuite.scala

```diff
@@ -270,7 +267,9 @@ abstract class KafkaRelationSuiteBase extends QueryTest with SharedSparkSession
   test("no matched offset for timestamp - startingOffsets") {
     val (topic, timestamps) = prepareTimestampRelatedUnitTest
 
-    val e = intercept[SparkException] {
+    // KafkaOffsetReaderConsumer and KafkaOffsetReaderAdmin both throws AssertionError
```

Comment:
In order to help reviewers I've created a small extract how the exception difference look like.
With `UninterruptibleThread` test execution:
```
org.apache.spark.SparkException(Exception thrown in awaitResult: )
java.util.concurrent.ExecutionException(Boxed Error)
java.lang.AssertionError(assertion failed: No offset matched from request of topic-partition topic-0-2 and timestamp 9223372036854775807.)
```

Without `UninterruptibleThread` test execution:
```
java.lang.AssertionError(assertion failed: No offset matched from request of topic-partition topic-0-2 and timestamp 9223372036854775807.)
```